django/db/migrations/autodetector.py

```diff
@@ -1458,8 +1484,13 @@ def create_altered_constraints(self):
 
             old_constraints = old_model_state.options[option_name]
             new_constraints = new_model_state.options[option_name]
-            add_constraints = [c for c in new_constraints if c not in old_constraints]
-            rem_constraints = [c for c in old_constraints if c not in new_constraints]
+
+            add_constraints = self.compare_altered_constraints(
+                new_constraints, old_constraints
+            )
+            rem_constraints = self.compare_altered_constraints(
+                old_constraints, new_constraints
+            )
```

Comment:
Thanks for this patch, however (IMO) it's not an acceptable solution. First of all, because you're basically mocking the `__eq__` operator (and deconstruction) for constraints and migrations must be able to recreate the current state of the models only from the migration operations. With this patch it's no longer possible.

Maybe we could introduce a new no-op operation, e.g. `AlterConstraint` that could be used instead of `Remove`/`Add` round-trip when an attribute doesn't impact DDL definitions, I'm just not sure it's worth doing.

Reply:
thanks @felixxm for advices

Reply:
Can I ask you how do I check if an attribute doesn't impact DDL definitions without using `__eq__` operator? @felixxm

Reply:
> Can I ask you how do I check if an attribute doesn't impact DDL definitions without using `__eq__` operator? @felixxm

We could add an internal hook for this, something similar to the `_field_should_be_altered()`.

Reply:
@felixxm I will implement something like that. Do I have to assign the ticket to myself?